### PR TITLE
Add an exclusion list to statistics, in order to exclude IP addresses from statistics.

### PR DESCRIPTION
**Pull request by vogje01**
This is a small extention to the Stats package, which allows to define an exclusion list containing IP addresses ot network subnet masks from entering the statistics. The extension adds a new tabs to the statistics settings admin page, which allows to add/delete IP addresses/subnet masks from the exclusion list. All request IP addresses matching the defined entries will not enter in the statistics. 

A new table has been added: '<prefix>_stats_exclude_ip' containing id,ip_address,remarks,username,timestamp.

A small extension is also added to the tabs configuration, so that after a commit (add/delete) the current tab is reselected.

Screenshot:
exclude_list

### CODE REVIEW

File: core_modules/Stats/Controller/Counter.php [hunk 1]
```diff
@@ -142,72 +142,94 @@ public function __construct($arrRobots, $arrBannedWords, $cx)
...
-                        // count browser
-                        if ($this->arrConfig['count_browser']['status']) {
-                            $this->_countBrowser();
-                        }
-
-                        // count operating system
-                        if ($this->arrConfig['count_operating_system']['status']) {
-                            $this->_countOperatingSystem();
+                    // Check exclude list
+                    if(!$this->_skipIps()) {
```
Comment (hbdsklf) [inline core_modules/Stats/Controller/Counter.php]: Missing argument $ip

File: core_modules/Stats/Controller/Stats.class.php [hunk 2]
```diff
@@ -1116,6 +1116,52 @@ function _xmlCountryEndTag($parser, $name){
...
+        $query = "DELETE FROM ".DBPREFIX."stats_exclude_ip WHERE id=".$id;
+        $objDatabase->Execute($query);
+    }
+
+    /**
+     * Delete a IP address from the exclusion list.
+     *
+     * @param ip IP address to delete.
+     */
+    function _delIpsFromExclusionList($ids) {
```
Comment (hbdsklf) [inline core_modules/Stats/Controller/Stats.class.php]: Either drop all IDs at once (performance, preferred way) or call _delIpsFromExclusionList()

File: core_modules/Stats/Controller/Stats.class.php [hunk 3]
```diff
@@ -1116,6 +1116,52 @@ function _xmlCountryEndTag($parser, $name){
...
+     * @param ip IP address to delete.
+     */
+    function _delIpFromExclusionList($id) {
+        global $objDatabase;
+        $query = "DELETE FROM ".DBPREFIX."stats_exclude_ip WHERE id=".$id;
+        $objDatabase->Execute($query);
+    }
+
+    /**
+     * Delete a IP address from the exclusion list.
```
Comment (hbdsklf) [inline core_modules/Stats/Controller/Stats.class.php]: Wrong method (and param) description.

File: core_modules/Stats/Controller/Stats.class.php [hunk 4]
```diff
@@ -1198,6 +1272,30 @@ function _showSettings() {
             'STATS_SETTINGS_PAGING_LIMIT'        => $this->arrConfig['paging_limit']['value'],
             'STATS_SETTINGS_PAGING_LIMIT_VISITOR_DETAILS'    => $this->arrConfig['paging_limit_visitor_details']['value'],
         ));
+
+        // Initialite exclude list
```
Comment (hbdsklf) [inline core_modules/Stats/Controller/Stats.class.php]: Typo: "initialize"

**Comment by hbdsklf** [comment]
Hi vogje01,

Thank you very much for your contribution!

I added some comments in the code. Additional to those there are some general points I'd like to address:

- Please escape all MySQL queries with back-ticks
- It would be nice to "return early" (especially in Counter.php), more info on this can be found here: https://pear.php.net/manual/en/standards.bestpractices.php

Please tell me if you agree and if you feel ready to fix those points. We'd like to support you in order to integrate this feature soon!

Best regard

hbdsklf

**Comment by vogje01** [comment]
asd

**Pull request closed without merging**